python/az/aro/azext_aro/_aad.py

```diff
@@ -54,6 +56,13 @@ def get_service_principal(self, app_id):
         return None
 
     def create_service_principal(self, app_id):
-        return self.client.service_principals.create(ServicePrincipalCreateParameters(
-            app_id=app_id,
-        ))
+        retries = 0
+        while True:
+            try:
+                return self.client.service_principals.create(
+                    ServicePrincipalCreateParameters(app_id=app_id))
+            except GraphErrorException:
+                if retries >= 3:
+                    raise
+                retries += 1
+                time.sleep(10)
```

Comment:
one more ask - can we log this?  check the other az code, I think it's something like
```
from knack.log import get_logger
# at module level
logger = get_logger(__name__)
logger.debug('blah')
```

Reply:
I thought it might be useful to also log what retry it is on and how many are left, but if that's too much we can just log the exception